tsl/test/sql/compression_errors.sql

```diff
@@ -469,3 +469,18 @@ SELECT compress_chunk(show_chunks('readings'), TRUE);
 INSERT INTO readings("time", candy, battery_temperature)
     VALUES ('2022-11-11 11:11:11', 33, 0.3)
 ;
+
+-- Segmentby checks should be done for unique indexes without
+-- constraints, so create a table without constraints and add a unique
+-- index and try to create a table without using the right segmentby
+-- column.
+CREATE TABLE table_unique_index(
+       location smallint not null,
+       device_id smallint not null,
+       time timestamptz not null,
+       value float8 not null
+);
+CREATE UNIQUE index ON table_unique_index(location, device_id, time);
+SELECT table_name FROM create_hypertable('table_unique_index', 'time');
+ALTER TABLE table_unique_index SET (timescaledb.compress);
```

Comment:
```suggestion
-- Will warn because the lack of segmentby/orderby compression options
ALTER TABLE table_unique_index SET (timescaledb.compress);
ALTER TABLE table_unique_index SET (timescaledb.compress = off);
-- Will enable compression without warnings
ALTER TABLE table_unique_index SET (timescaledb.compress, timescaledb.compress_segmentby = 'location', timescaledb.compress_orderby = 'device_id');
```
What about extend a bit this regression test?

Reply:
Added these and a few extra. PTAL.